### ISSUE TITLE
Variance annotations are not allowed in type aliases for intersections

### ISSUE THREAD

**Issue by Andarist** [microsoft/TypeScript#59561]
### 🔎 Search Terms

variance measurement markers sub sup intersection type aliases

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.6.0-dev.20240808#code/C4TwDgpgBAKgYgewQHgGoEMA2BXCA+KAXigG8oFtgAuKDHaAXwG4AoUSWAIXQCc0tcBYmQCWAOxoAKXgHMadXAEoiBAG4IRAEyjM24aDG4AvZOPKVaA-EViIUC6wDIuvfvTysgA

### 💻 Code

```ts
interface Foo<Value> {
  out: Value;
}
interface Bar<Value> {
  in: (arg: Value) => void;
}

// }
interface Foo<Value> {
  out: Value;
}
interface Bar<Value> {
  in: (arg: Value) => void;
}

type Baz<in out Value> = Foo<Value> & Bar<Value>;

type First<in out Value> = Foo<Value>;
type Second<in out Value> = Bar<Value>;

// Variance annotations are only supported in type aliases for object, function, constructor, and mapped types.(2637)
type Baz<in out Value> = Foo<Value> & Bar<Value>;

// both members are allowed to have annotations
type First<in out Value> = Foo<Value>;
type Second<in out Value> = Bar<Value>;
```

### 🙁 Actual behavior

Error is raised

### 🙂 Expected behavior

I'd expect this to work, especially since https://github.com/microsoft/TypeScript/pull/56418 allowed it to work on more types. Note that this issue isn't even strictly related to using generic references as constituent members. The same issue happens with type aliases and interfaces

### Additional information about the issue

Maybe this is expected but I couldn't quickly find any issue related to it or an explicit mention of intersections being purposefully disallowed here

**Comment by RyanCavanaugh** [comment]
The behavior matches the error message; why is this unexpected?

**Comment by Andarist** [comment]
I'm not saying it doesn't match the error message. That part is fine - although somebody could argue that `{ a: 1 } & { b: 2 }` is still an object type. I just expected an intersection like this to almost always be substitutable with its "resolved" form and it caught me off guard that it wasn't here. So I got interested in why this isn't allowed today or why it might no be a good idea to allow it 😉

**Comment by RyanCavanaugh** [comment]
"Is an intersection of object types an intersection, an object type, or both" the greatest thread in the history of forums, locked by a moderator after 12239 pages of heated debate

**Comment by RyanCavanaugh** [comment]
The real answer IIRC is that variance is never used when checking intersections, but I might be off here. @ahejlsberg ?

**Comment by Andarist** [comment]
FWIW, I checked what happens under the hood. When comparing 2 instantiations of `Baz` from this example `relateVariances` gets called as normal and in the case of failure `allowStructuralFallback` gets computed as `false`. So this intersection takes the regular variance-based codepath as far as I can tell.
